### DISCUSSION TITLE
scan.py example showing only one peripheral

### DISCUSSION THREAD

**Discussion by gibran404** [simpleble/simpleble#151] [Q&A] [answered]
when i run the scan example through simplepyble, it doesnt show any actual devices in the range and instead always shows this one set of details which is some non existant device.
the details it returns are as follows:

      [19:ef:e2:46:5f:6f] - Non-Connectable
          Address Type: BluetoothAddressType.RANDOM
          Tx Power: -32768 dBm
          Manufacturer ID: 6
          Manufacturer data: b'\x01\t \x02\xb3\xae\xc3\x7fJ\xa2\xf6\xdb\x9d+i\xbe\x1c&\xaf.\x86qlzN\x99\x07'

the address doesn't match any bluetooth device in the device manager and neither does such a device exist in the bluetooth range.
am i doing something wrong or is it some bug or incompatibility issue?

**Comment by kdewald** [comment, marked as answer]
Hey @zalva404, thanks for reaching out.

Just for context, could you specify what version of `simplepyble` you're using and on what operating system?

Additionally, there are a few points to highlight here:
- Devices that appear in the device manager are not necessarily broadcasting at the exact time you're scanning. The OS will show you everything that you've connected in the past.
- The result that you're getting is valid. Bluetooth devices will sometimes broadcast in non-connectable mode to give away their presence or broadcast some piece of data. I can see from your example that it's a Microsoft device (based on the Manufacturer ID).

Based on this I would say that the library is working as intended, you just need to review your test setup to be sure that you'll get the output you desire.

**Reply by gibran404** [reply]
im running the 0.5.0 version of simplepyble, on windows 10 19044 build. but I've also tried 0.6.0 dev2 and the result is the same.

previously it was showing only one device regardless of other devices being in range or not. But this one device has somehow disappeared as well now and it doesn't show any devices in the scan.

what I suspect it was previously showing was a driver for a virtual machine, but if that was the case, it should still be here since I have not removed any of the virtual machines. 
Although I'm not sure if it makes any difference but it is interesting to note that when the virtual machine is running, it steals the Bluetooth from the main windows OS and the bluetooth becomes unavailable on the main OS.
